travis/setup_dependencies_common.sh

```diff
@@ -37,25 +37,69 @@ function retry_on_known_error() {
         echo "ERROR: Function retry_on_known_error() called without arguments." 1>&2
         return 1
     fi
+    _tmp_output_file="tmp.txt"
     _n_retries=0
     _exitval=0
     _retry=true
     while $_retry; do
         _retry=false
-        # Execute the wrapped command and get its unified output:
-        _output=$($@ 2>&1)
+        # Execute the wrapped command and get its unified output.
+        # This command needs to run in the current shell/environment in case
+        # it sets environment variables (like 'conda install' does)
+        #
+        # tee will both echo output to stdout and save it in a file. The file
+        # is needed for some error checks later. Output to stdout is needed in
+        # the event a conda solve takes a really long time (>10 min). If
+        # there is no output on travis for that long, the job is cancelled.
+        set +e
```

Comment:
wouldn't this be littering the logs too much?

Reply:
I think `set +e` just tells the script not to exit if it hits an error without affecting the verbosity of the output.

The `cat` below this could be  wrapped in an "if DEBUG" but I think it is better if the output of the conda command shows up in the log.